tests/php/test-fieldmanager-context-menuitem.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/**
+ * Tests the Menu Item context
+ *
+ * @group context
+ * @group menuitem
+ */
+class Test_Fieldmanager_Context_Menuitem extends WP_UnitTestCase {
+	public function setUp() {
+		parent::setUp();
+		Fieldmanager_Field::$debug = true;
+
+		$this->post = array(
```

Comment:
This would be simplified greatly by doing `self::factory()->post->create_and_get(` which obviates the need for the `get_post` below, and handles the post insertion. It will also randomize the post title and content for you, so you can eliminate that. In a few places below, you're already using `$this->post->ID` so you can just do that instead of `$this->post_id`.